### PR TITLE
New features

### PR DESCRIPTION
**Pull request by simeon-walker**
Hi,
I've added a couple of features to flick-download:

1. If the photo has no title then the Flickr ID will be used for the filename.
2. Date based folders can be created for the file being downloaded.
3. File times are only updated to match the photo timestamp when requested.

If you like the changes then please merge.
Regards,
Simeon

### CODE REVIEW

**Comment by beaufour** [comment]
Thanks for the patch! I like the feature addition. I'll try to look at the code this weekend. There are two style fixes that should be fixed at least:

    flickr_download/flick_download.py:203:101: E501 line too long (108 > 100 characters)
    flickr_download/flick_download.py:205:101: E501 line too long (114 > 100 characters)

See: https://travis-ci.org/beaufour/flickr-download/jobs/50960495

**Comment by simeon-walker** [comment]
I've fixed the line length issue.

File: flickr_download/flick_download.py [hunk 1]
```diff
@@ -169,6 +178,10 @@ def main():
                         default=None, help='Quality of the picture')
     parser.add_argument('-n', '--naming', type=str, metavar='NAMING_MODE',
                         default='title', help='Photo naming mode')
+    parser.add_argument('-f', '--folder_naming', type=str, metavar='FOLDER_NAMING',
+                        default='%Y/%m/%d', help='Photo folder naming')
+    parser.add_argument('-ft', '--file_times', action='store_true',
+                        default=False, help='Set file timestamp to photo taken timestamp')
```
Comment (beaufour) [inline flickr_download/flick_download.py]: I'm happy with the feature, but you are changing the default to `False` here. For backwards compatibility you should use `True`

I would also suggest `Set file time to time when the photo was taken` as the help text

File: flickr_download/flick_download.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 from dateutil import parser
 import yaml
 
-from flickr_download.filename_handlers import get_filename_handler
+from flickr_download.filename_handlers import get_filename_handler, get_foldername
```
Comment (beaufour) [inline flickr_download/flick_download.py]: I try to do one import per line

File: flickr_download/filename_handlers.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 from collections import defaultdict
+from dateutil import parser
 
 
 def title(pset, photo, suffix):
```
Comment (beaufour) [inline flickr_download/filename_handlers.py]: you need to amend the docstring to say that it falls back to id

File: flickr_download/filename_handlers.py [hunk 4]
```diff
@@ -18,7 +19,10 @@ def title(pset, photo, suffix):
     @param suffice: str, optional suffix
     @return: str, the filename
     """
-    return '{0}{1}.jpg'.format(photo.title, suffix)
+    if photo.title:
+        return '{0}{1}.jpg'.format(photo.title, suffix)
+    else:
```
Comment (beaufour) [inline flickr_download/filename_handlers.py]: you don't need an `else` after a `return`

File: flickr_download/filename_handlers.py [hunk 5]
```diff
@@ -81,3 +85,13 @@ def get_filename_handler(name='title'):
     @return: Function, handler
     """
     return HANDLERS[name]
+
+
+def get_foldername(photo, pattern):
+    """
+    Returns date based foldername
```
Comment (beaufour) [inline flickr_download/filename_handlers.py]: needs docstring for arguments and return value. Also this file is for functions that determine the file names, put it in the mail `.py` file

File: flickr_download/flick_download.py [hunk 6]
```diff
@@ -169,6 +178,10 @@ def main():
                         default=None, help='Quality of the picture')
     parser.add_argument('-n', '--naming', type=str, metavar='NAMING_MODE',
                         default='title', help='Photo naming mode')
+    parser.add_argument('-f', '--folder_naming', type=str, metavar='FOLDER_NAMING',
+                        default='%Y/%m/%d', help='Photo folder naming')
```
Comment (beaufour) [inline flickr_download/flick_download.py]: it is not entirely clear from the doc string what this does, could you clarify. It also, defaults to it being on. It needs to be off for backwards compatibility

**Comment by beaufour** [comment]
I added the falling back to the id if there is no title just now. Let me know if you have any comment on the rest?

**Comment by ustcscgy** [comment]
Run into the "album title has '/' in it, so can't 'os.mkdir(pset.title)' " issue, merge this pull request, please?

**Comment by beaufour** [comment]
@ustcscgy I have a fair amount of unaddressed comments on it I'm afraid, but I'll see if I can't sneak in a fix for the '/' album title in the not so distant future :)

**Comment by ustcscgy** [comment]
Thank you for your reply, I have hacked it locally on my machine, by replacing `pset.title` with `pset.title.replace("/","_")`.

**Comment by beaufour** [comment]
@ustcscgy I added a fix in v0.2.11

**Comment by ustcscgy** [comment]
Thank you.

**Comment by beaufour** [comment]
I added some of the features, the rest still has unaddressed comments. Closing as it is now quite old.

**Pull request closed without merging**